### ISSUE TITLE
 Display HTTP 4xx Error Code if it occurs (eg if upload in webapp fails)

### ISSUE THREAD

**Issue by til2000** [photoprism/photoprism#3155]
#### 1. What is not working as documented?

Original issue: `uploads via webapp not possible`
Original symptoms: 
```
Uploading Pictures via web app (three dot menu -> upload cloud icon) 
to self hosted docker compose :latest package
when executing import for 2 photos from iOS (ipad, iphone) any browser (safari, chrome, firefox) 
    --> regular indexing happens but nothing is being imported, no error, no success. 
```

**Root Cause in my case**:
Nginx default configuration allows 1m post upload. After `client_max_body_size 64M;` 
is configured, all works fine.

This leads to:  What is not working as documented?

**Photoprism isn't reacting to HTTP 413 error.**

#### 2. How can we reproduce it?

1. Go to example.com/photoprism
2. Click on 'three dots' -> 'cloud with upload arrow' -> 'upload cloud with arrow'
3. Chose a picture larger than configured client_max_body_size  -> click on 'add'
4. Wait and see that the processing happens without showing an error
5. Check Search / Review and not find the uploaded pictures

#### 3. What behavior do you expect?

upload fails with information that file is too large

**Comment by lastzero** [comment]
> 1. What is not working as documented?

(a) Failed uploads are literally the first thing mentioned when you read the corresponding documentation: https://docs.photoprism.app/getting-started/proxies/nginx/

(b) While it would be possible to add handlers for all kinds of proxy errors and other problems, this is not something we promise our users as a feature they can rely on. In fact, we provide a NSFW filter that blocks unsafe uploads. So it's an expected behavior that not everything can be uploaded.

(c) Our development preview includes frontend improvements so that you see an "upload failed" message when all files have failed. Adding more messages would be an additional improvement, but would require translations for each case. This is clearly not a solution that can be provided in response to a bug report.

(d) As a workaround, we recommend that you do not use NGINX or configure it to meet your needs, being aware of what restrictions you have enabled.

**Comment by til2000** [comment]
> (c) Our development preview includes frontend improvements so that you see an "upload failed" message when all files have failed. Adding more messages would be an additional improvement, but would require translations for each case. This is clearly not a solution that can be provided in response to a bug report.

Thanks for the info! I am an excited user of the software and admire all effort brought towards it.
I'm not able to go further to download dev editions. 

As end user I hope to find helpful information via google, and none of the existing issues about client_max_body_size came up when I was searching for the issue at hand: Upload failed when using webapp upload.

The underlying problem imho is, that I haven't been informed about the HTTP error code.
This will be an issue as well for nsfw content - if users want to upload it and users expect it to upload and it doesn't, then there will be issues raised, which could be prevented by providing the error code (be it 413 for too large, or 403 for 'Forbidden (due to nsfw)' )

If for whatever reason it should be unknown that nsfw content is being filtered, this should be configurable - generally, as admin and user I would like to have as much transparency as possible, as this prevents extra work for everybody involved (except the developers)

at this point again big kudos to everybody from product side - it's a great tool and I appreciate your work a lot !

**Comment by til2000** [comment]
After reading your point again, I'd like to further narrow down the title. 
I do believe that a picture specific information could be provided with minimal translation effort.
In case a specific picture wasn't processed, the HTTP Error code would suffice as response.

**Comment by lastzero** [comment]
Well, nginx is so hard to configure, they have a dedicated mistakes page that is longer than the complete documentation of most other software: https://www.nginx.com/resources/wiki/start/topics/tutorials/config_pitfalls/

So even if you get the uploads right, you will hit the next issue. We can't help every user with configuring this third party product when there are much easier to use alternatives we actively recommend. 

You should already be informed about nsfw rejections as that's an expected problem related to our own software.

**Comment by lastzero** [comment]
Of course, I fully agree that helping users troubleshoot server-side issues is a worthwhile thing to do. However, in the case of NGINX, it would be symptom-fighting, as this particular proxy is not well-suited for end users. For example, if you've successfully uploaded large videos, you may find that streaming doesn't work as expected.

So if we ever have time to work on the upload (which is very difficult given all the other tasks and requests), I would rather spend that time on drag & drop or ZIP support - unless we can grow our team, which ultimately depends on how many end users are willing to pay for the software they use.
